Makefile

```diff
@@ -33,17 +35,22 @@ clean:
 
 generate-accounts:
 # Normal mail accounts
-	@ docker run --rm -e MAIL_USER=user1@localhost.localdomain -e MAIL_PASS=mypassword -t $(NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)"' > test/config/postfix-accounts.cf
-	@ docker run --rm -e MAIL_USER=user2@otherdomain.tld -e MAIL_PASS=mypassword -t $(NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)"' >> test/config/postfix-accounts.cf
-	@ docker run --rm -e MAIL_USER=user3@localhost.localdomain -e MAIL_PASS=mypassword -t $(NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)|userdb_mail=mbox:~/mail:INBOX=~/inbox"' >> test/config/postfix-accounts.cf
+	@ docker run --rm -e MAIL_USER=user1@localhost.localdomain -e MAIL_PASS=mypassword -t $(IMAGE_NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)"' > test/config/postfix-accounts.cf
+	@ docker run --rm -e MAIL_USER=user2@otherdomain.tld -e MAIL_PASS=mypassword -t $(IMAGE_NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)"' >> test/config/postfix-accounts.cf
+	@ docker run --rm -e MAIL_USER=user3@localhost.localdomain -e MAIL_PASS=mypassword -t $(IMAGE_NAME) /bin/sh -c 'echo "$$MAIL_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MAIL_USER -p $$MAIL_PASS)|userdb_mail=mbox:~/mail:INBOX=~/inbox"' >> test/config/postfix-accounts.cf
 	@ echo "# this is a test comment, please don't delete me :'(" >> test/config/postfix-accounts.cf
 	@ echo "           # this is also a test comment, :O" >> test/config/postfix-accounts.cf
 
 # Dovecot master accounts
-	@ docker run --rm -e MASTER_USER=masterusername -e MASTER_PASS=masterpassword -t $(NAME) /bin/sh -c 'echo "$$MASTER_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MASTER_USER -p $$MASTER_PASS)"' > test/config/dovecot-masters.cf
+	@ docker run --rm -e MASTER_USER=masterusername -e MASTER_PASS=masterpassword -t $(IMAGE_NAME) /bin/sh -c 'echo "$$MASTER_USER|$$(doveadm pw -s SHA512-CRYPT -u $$MASTER_USER -p $$MASTER_PASS)"' > test/config/dovecot-masters.cf
 
-tests:
-	@ NAME=$(NAME) ./test/bats/bin/bats --timing test/*.bats
+tests: test/part/0 test/part/1 test/part/2 test/part/3
+
+test/part/%:
+# part/0 => tests run in a serialized manner
+# part/x where (x > 0) => tests are run in parallel
+	@ if [[ $* -eq 0 ]]; then ./test/bats/bin/bats --timing test/serial.*.bats; else \
+		./test/bats/bin/bats --timing --jobs $(PARALLEL_JOBS) test/parallel.$*.*.bats; fi
```

Comment:
Not a huge fan of the change here.

You've got a conditional to handle serial vs parallel testing, and supporting commentary to further clarify that.

Let's change this to `test/part/0:` and after that `test/part/%`, I assume it's smart enough to only match and handle numbers after `0`, if not, we can just have clear commands `tests-serial` + `tests-parallel`, both called by `make tests`. I think that would be clearer?

Can you clarify if I understand this correctly? You are running `parallel` test files in parallel, but not their individual test cases within a file correct? Default parallel jobs is two, so for local users the resources required may be twice what was needed before? I think that was mostly only an issue for ClamAV which is now it's own serial test, thus probably nothing to worry about.

If a user does want to run tests all in serial, then setting `PARALLEL_JOBS=1` will do the trick?

Reply:
This part could get some rework. Depending on your rebase I will tackle it afterwards.

The docs preview explains the new strategy better: <https://pullrequest-2822--dms-doc-previews.netlify.app/contributing/general/#understanding-the-test-suite>. Locally, individual test files are run in parallel, but only those that have a `parallel` in their name.

---

> If a user does want to run tests all in serial, then setting `PARALLEL_JOBS=1` will do the trick?

Exactly.